### DISCUSSION TITLE
Fake_processor doesn't support merchant

### DISCUSSION THREAD

**Discussion by berteotti** [pay-rails/pay#864] [Ideas]
We are using `Company` model with `merchant_processor` linked to stripe and we calling `merchant_processor` `onboarding_complete` in our tests.

If we use `stripe` as processor it will call stripe server which it’s not what we want.

But if we use `fake_processor` as processor tests fail because `fake_processor` doesn’t have a `merchant`.

Any idea on how can we fix our tests? 

Any plans to add `merchant` to `fake_processor`?

**Comment by excid3** [comment]
We haven't implemented a fake merchant processor yet, but we will eventually.

**Comment by Diogomartf** [comment]
Would you be interested in getting help on that? 

If you give us some guidance we might be able to add that functionallity.

**Comment by excid3** [comment]
Sure, although I don't know exactly what needs to be done. For now it probably would need to just mimic Stripe and just generate Merchant record.

**Comment by Diogomartf** [comment]
Ok, we will try to do that. 👍
